### DISCUSSION TITLE
How to set up TRPC with NextJS 13 App Router

### DISCUSSION THREAD

**Discussion by eden-chan** [trpc/trpc#4680] [General]
From a similar discussion, it does not seem TRPC is nextjs 13 compliant for HOC: (please correct me if I'm mistaken) 
https://github.com/trpc/trpc/discussions/3185

https://youtu.be/YQMSietiFm0?t=870 
Layout replaces App and Document in NextJS 13
```Layout.tsx

import './globals.css'
import type { Metadata } from 'next'
import { Inter } from 'next/font/google'
 
 
const inter = Inter({ subsets: ['latin'] })
 
export const metadata: Metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}
 
 
export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en">
      <body className={inter.className}>{children}</body>
    </html>
  )
}
 
```

https://github.com/trpc/trpc/discussions/3185#discussioncomment-5628995 
"Correct me if i am wrong but instead of using HOC, direct client or caller can be used as mentioned below" https://trpc.io/docs/server/server-side-calls 

https://trpc.io/docs/client/nextjs/setup
<img width="892" alt="image" src="https://github.com/trpc/trpc/assets/60107091/44114a2e-8884-4ae7-87e8-c911e3146ad9">

**Comment by curiouscod3** [comment]
https://github.com/trpc/trpc/tree/main/examples/.experimental/next-app-dir
https://github.com/trpc/examples-next-app-dir/tree/main
